### PR TITLE
Patterns: Add skeleton of new `Patterns` page

### PR DESCRIPTION
**Pull request by stephanethomas**
This pull request adds a new `Patterns` page at http://calypso.localhost:3000/patterns. This is the very first step in introducing a patterns library on WordPress.com. This page is rendered client-side (as usual) but also server-side (for SEO and performance purposes). Its layout differs depending on whether the user is logged in or not, even if ultimately its content will be the same:

Logged Out | Logged In
------ | -----
Logged Out | Logged In

#### Additional notes

I went for two distinct entry points `index.node.js` and `index.web.js` as localization will be handled differently client-side and server-side through [distinct middleware](https://github.com/Automattic/wp-calypso/pull/87511/files#diff-ddc57c32c80226276fa5ec0199e3a02eb4113ddc9779252edd57d720d72ad506). Here are a few examples of pages rendered server-side, with varying degrees of support:

* http://calypso.localhost:3000/tags
* http://calypso.localhost:3000/themes/filter/business
* http://calypso.localhost:3000/themes
* http://calypso.localhost:3000/site-profiler
* http://calypso.localhost:3000/plugins

Note how different they look depending on whether the user is logged in or not. We'll have to decide how far we want to go in regard to that: should we show the masterbar when logged in? Should we show the sidebar? Right now there is a very annoying flicker effect when loading the page since the version rendered first by the server has a different layout than the one rendered by the client.

#### Testing instructions

1. Run `git checkout add/patterns-page` and start your server, or access a [live branch](https://github.com/Automattic/wp-calypso/pull/87647#issuecomment-1954059649)
2. Open the [`Patterns` page](http://calypso.localhost:3000/patterns) while logged out
3. Assert that it looks like the screenshot above
4. Open that page again while logged in
5. Assert that it looks like the screenshot above

### CODE REVIEW

File: client/my-sites/patterns/patterns.tsx [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+export default () => {
+	return (
+		<Main fullWidthLayout isLoggedOut>
+			<div>
+				<h1>Build your perfect site with patterns</h1>
+
+				<p>
+					Hundreds of expertly designed, fully responsive patterns allow you to craft a beautiful
+					site in minutes.
+				</p>
```
Comment (stephanethomas) [inline client/my-sites/patterns/patterns.tsx:14]: Those strings are not localized on purpose - it's too early for this.

File: packages/help-center/src/hooks/use-zendesk-messaging.ts [hunk 2]
```diff
@@ -50,6 +50,7 @@ export default function useZendeskMessaging(
 	}, [ setMessagingScriptLoaded, enabled, zendeskKey, isMessagingScriptLoaded ] );
 
 	if (
+		document &&
```
Comment (stephanethomas) [inline packages/help-center/src/hooks/use-zendesk-messaging.ts:53]: This is unrelated to my changes, but this fixes an error triggered when loading the `Plugins` [page](http://calypso.localhost:3000/plugins):

```javascript
ERROR calypso: document is not defined (reqId=cfe59271-5514-4a6c-bb65-d9e7643eaa5c, url=/plugins, env=development, userAgent="Chrome 121", path
=/plugins)
    ReferenceError: document is not defined
        at useZendeskMessaging (/var/sources/build/webpack:/wp-calypso/packages/help-center/src/hooks/use-zendesk-messaging.ts:52:2)
        at usePresalesChat (/var/sources/build/webpack:/wp-calypso/client/lib/presales-chat/index.ts:70:38)
        at PluginsBrowser (/var/sources/build/webpack:/wp-calypso/client/my-sites/plugins/plugins-browser/index.jsx:101:2)
        at renderWithHooks (/var/sources/build/node_modules/react-dom/cjs/react-dom-server-legacy.node.development.js:5662:16)
        at renderIndeterminateComponent (/var/sources/build/node_modules/react-dom/cjs/react-dom-server-legacy.node.development.js:5735:15)
        at renderElement (/var/sources/build/node_modules/react-dom/cjs/react-dom-server-legacy.node.development.js:5950:7)
        at renderNodeDestructiveImpl (/var/sources/build/node_modules/react-dom/cjs/react-dom-server-legacy.node.development.js:6108:11)
        at renderNodeDestructive (/var/sources/build/node_modules/react-dom/cjs/react-dom-server-legacy.node.development.js:6080:14)
        at renderNode (/var/sources/build/node_modules/react-dom/cjs/react-dom-server-legacy.node.development.js:6263:12)
        at renderHostElement (/var/sources/build/node_modules/react-dom/cjs/react-dom-server-legacy.node.development.js:5646:3)
        at renderElement (/var/sources/build/node_modules/react-dom/cjs/react-dom-server-legacy.node.development.js:5956:5)
        at renderNodeDestructiveImpl (/var/sources/build/node_modules/react-dom/cjs/react-dom-server-legacy.node.development.js:6108:11)
        at renderNodeDestructive (/var/sources/build/node_modules/react-dom/cjs/react-dom-server-legacy.node.development.js:6080:14)

```

Comment (fredrikekelund) [reply packages/help-center/src/hooks/use-zendesk-messaging.ts:53]: Were we generating that on every page load? 😓

**Review by fredrikekelund — APPROVED** [review]
Great start! LGTM 👍

> should we show the masterbar when logged in?

I think that kinda makes sense. That's also what we do on wordpress.com/support, which might be a good reference because it renders basically the same for both logged-out and logged-in users. Masterbar on top and then the "logged-out header" below.

> Should we show the sidebar?

I assume you mean the Calypso sidebar. If so, we opted not to do that in the project thread pcYYuD-1eY-p2

**Pull request merged**